File: src/main/java/com/saomc/social/friends/FriendRequest.java
package com.saomc.social.friends;

import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public final class FriendRequest {

    private final String friendName;
    public int ticks;

    public FriendRequest(String name, int maxTicks) {
        friendName = name;
        ticks = maxTicks;
    }

    private boolean equals(FriendRequest request) {
        return equals(request == null ? null : request.friendName);
    }

    public final boolean equals(String name) {
        return friendName.equals(name);
    }

    public final boolean equals(Object object) {
        return object instanceof FriendRequest ? equals((FriendRequest) object) : equals(String.valueOf(object));
    }

}

File: src/main/java/com/saomc/saoui/social/friends/FriendRequest.kt
package com.saomc.saoui.social.friends

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly

@SideOnly(Side.CLIENT)
class FriendRequest(private val friendName: String, var ticks: Int) {

    fun equals(name: String?): Boolean {
        return friendName == name
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FriendRequest

        if (friendName != other.friendName) return false

        return true
    }

    override fun hashCode(): Int {
        return friendName.hashCode()
    }


}
